pkg/apis/serving/v1beta1/inference_service_validation.go

```diff
@@ -252,9 +261,19 @@ func validateInferenceServiceAutoscaler(isvc *InferenceService) error {
 	return nil
 }
 
+// Validate of autoscaler KEDA metrics
+func validateKEDAMetrics(metric ScaleMetric) error {
```

Comment:
In Go 1.18 and later, you can use the slices package from the standard library to check if an item exists in a list (slice), so you can use
```
	if slices.Contains(constants.AutoscalerAllowedHPAMetricsList, constants.AutoscalerMetricsType(metric)) {
		return nil
	}
```
to check if constants.AutoscalerMetricsType(metric) is in the array.